File: src/main/java/com/core/base/corebase/domain/common/BaseDateTimeEntity.java
package com.core.base.corebase.domain.common;

import jakarta.persistence.Column;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.MappedSuperclass;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Getter
@Setter
@MappedSuperclass
@EntityListeners(AuditingEntityListener.class)
public abstract class BaseDateTimeEntity {

    @CreatedDate
    @Column(updatable = false)
    private LocalDateTime createdAt = LocalDateTime.of(1970, 1, 1, 0, 0, 1);

    @LastModifiedDate
    private LocalDateTime updatedAt = LocalDateTime.of(1970, 1, 1, 0, 0, 1);
}

File: src/main/kotlin/com/core/base/corebase/domain/common/BaseDateTimeEntity.kt
package net.huray.backend.minuting.entity.common

import jakarta.persistence.Column
import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@EntityListeners(AuditingEntityListener::class)
@MappedSuperclass
abstract class BaseDateTimeEntity {

    @CreatedDate
    @Column(updatable = false)
    var createdAt: LocalDateTime = LocalDateTime.of(1970, 1, 1, 0, 0, 1)!!

    @LastModifiedDate
    var updatedAt: LocalDateTime = LocalDateTime.of(1970, 1, 1, 0, 0, 1)!!

}
